src/parser/here_doc.c

```diff
@@ -39,37 +40,53 @@ static t_exit_code	append_line_to_heredoc(char *line,
 	return (SUCCESS);
 }
 
-static void	cleanup_here_doc(char *line, int fd)
+static int	cleanup_here_doc(char *line, int fd, char *file_name)
 {
+	int	*signal;
+	int	reopen_fd;
+
+	signal = heredoc_sigint();
 	close(fd);
 	free(line);
+	if (*signal)
+	{
+		*signal = 0;
+		reopen_fd = FILE_ERROR;
```

Comment:
Why do you need to set `reopen_fd = FILE_ERROR` ?

Reply:
This is so that the command that used the heredoc as input isn't executed. In the dispatcher we have an if statement that prevents commands with file errors of being performed.